core/src/main/java/com/linecorp/armeria/client/endpoint/healthcheck/AbstractHealthCheckedEndpointGroupBuilder.java

```diff
@@ -155,12 +162,53 @@ public AbstractHealthCheckedEndpointGroupBuilder withClientOptions(
         return this;
     }
 
+    /**
+     * Sets the maximum endpoint ratio of target selected candidates.
+     * @see PartialHealthCheckStrategyBuilder#maxEndpointRatio(double)
+     */
+    public AbstractHealthCheckedEndpointGroupBuilder maxEndpointRatio(double maxEndpointRatio) {
```

Comment:
- How about setting `null` to `partialHealthCheckStategyBuilder` if `maxEndpointRatio` is equal to 1.0?
- How about validating `maxEndpointRatio` before instantiating `PartialHealthCheckStrategyBuilder`, so that we do not enter the state where the builder stays in its default state? i.e.
  ```java
  checkArgument(maxEndpointRatio > 0 && maxEndpointRatio <= 1.0,
                "maxEndpointRatio: %s (expected: 0.0 < maxEndpointRatio < 1.0)",
                maxEndpointRatio);
  if (partialHealthCheckStrategyBuilder == null) { ... }
  ...
  ```

Reply:
> How about setting null to partialHealthCheckStategyBuilder if maxEndpointRatio is equal to 1.0?

How about this scenario? 
`Sets maxEndpointCount after sets maxEndpointRatio 1.0`
We've designed it's mutually exclusive. Should it be mutually exclusive even this scenario?

Reply:
> How about validating maxEndpointRatio before instantiating PartialHealthCheckStrategyBuilder, so that we do not enter the state where the builder stays in its default state?

fixed

Reply:
> How about this scenario? `Sets maxEndpointCount after sets maxEndpointRatio 1.0` We've designed it's mutually exclusive. Should it be mutually exclusive even this scenario?

That's a good point. How about setting 1.0 to `maxEndpointRatio` so that mutual exclusiveness is preserved? We could build `AllHealthCheckStrategy` when `maxEndpointRatio` is 1.0 in `build()`.